### PR TITLE
Fix #10

### LINKED ISSUE #10: Better to support Unicode scripts rather than blocks in `\p` and `\P`
I think it is better to support script names than block names for `\p` and `\P`.

Unicode block names are not supported by regexp implementations an interoperable way.  Block names as specified in

https://www.unicode.org/Public/UNIDATA/Blocks.txt

allow ASCII alpha-numerics, hyphens and spec.  Unicode says these names are supposed to be matched using

http://www.unicode.org/reports/tr44/#UAX44-LM3

(which says to ignore case, whitespace, underscore, hyphens, and any initial prefix string "is"). But implementations don't do this and compare names using a wide variety of different rules (as partially described by "Unicode blocks" section of https://www.regular-expressions.info/unicode.html).

Usage of block names can easily be replaced by ranges.  Furthermore, block names are not covered by the Unicode stability policy.

Scripts names don't have such a wide range of characters (there are also 4-char short script names) and scripts are generally preferable to blocks for regexps as explained here:

https://unicode.org/reports/tr18/#Character_Blocks

Script is in the minimal list of properties that Unicode requires but block is not

http://unicode.org/reports/tr18/#RL1.2

### PR DESCRIPTION
**Pull request by cabo**

### CODE REVIEW

**Review by glyn — APPROVED** [review]

File: draft-ietf-jsonpath-iregexp.md [hunk 1]
```diff
@@ -115,8 +115,8 @@ match `[^]`, which according to this grammar would parse as a
 positive character class containing the single character `^`.
 
 This is essentially XSD regexp without character class
-subtraction and multi-character escapes such as `\s`,
-`\S`, and `\w`.
+subtraction, without multi-character escapes such as `\s`,
+`\S`, and `\w`, and without Unicode blocks.
```
Comment (timbray) [inline draft-ietf-jsonpath-iregexp.md:119]: just curious why \S is being highlighted in dark red in GitHub diffs

File: draft-ietf-jsonpath-iregexp.md [hunk 2]
```diff
@@ -218,10 +218,10 @@ exceptions:
 
 * Not all regexp implementations
   support accesses to Unicode tables that enable
-  executing on constructs such as `\p{IsCoptic}`,
```
Comment (timbray) [inline draft-ietf-jsonpath-iregexp.md:221]: s/executing on constructs/executing constructs/ - the "on" isn't idiomatic

File: draft-ietf-jsonpath-iregexp.md [hunk 3]
```diff
@@ -218,10 +218,10 @@ exceptions:
...
+  executing on constructs such as `\p{Nd}`,
   although the `\p`/`\P` feature in general is now quite
   widely available. While in principle it’s possible to
-  translate these into codepoint-range matches, this also requires
+  translate these into character-class matches, this also requires
   access to those tables. Thus, regexp libraries in severely
   constrained environments may not be able to support I-Regexp
```
Comment (timbray) [inline draft-ietf-jsonpath-iregexp.md:226]: Is this still true with the removal of unicode blocks?

**Review by timbray — CHANGES_REQUESTED** [review]
1. Assuming you have used a tool for verifying the grammar? LGTM.
2. We are likely going to get the question "why not include unicode scripts?" Our answer is "not XSD compatible". Next question "Why is XSD compatibility important, just say XSD minus X plus Y?"  Personally I am in favor of leaving out the scripts because, as you pointed out, not really a mainstream use, but still, what's a good answer to that question?

File: draft-ietf-jsonpath-iregexp.md (hunk 1 above)
Comment (cabo) [reply draft-ietf-jsonpath-iregexp.md:119]: It is probably recognized as a syntax error, but I don't know for what syntax.

File: draft-ietf-jsonpath-iregexp.md (hunk 3 above)
Comment (cabo) [reply draft-ietf-jsonpath-iregexp.md:226]: Yes.  Turns out that my homework assignment proving this was done already:

https://github.com/CESNET/libyang/blob/87375f15159545a87a1e0de200f5d9d67e9091d7/src/schema_compile_node.c#L1023

This is a bit icky, as we don't have a \x or \u in iregexp, so you would have to include the actual unicode characters.

File: draft-ietf-jsonpath-iregexp.md (hunk 2 above)
Comment (cabo) [reply draft-ietf-jsonpath-iregexp.md:221]: Fixed in d249ec3

**Pull request merged**